src/node/desktop/src/main/detect_r.ts

```diff
@@ -37,116 +42,98 @@ function showRNotFoundError(msg: string): void {
  * @returns true if startup should continue, false on fatal error
  */
 export async function prepareEnvironment(): Promise<boolean> {
-  if (process.platform === 'win32') {
-    return await prepareEnvironmentWin32();
-  } else {
-    return await prepareEnvironmentPosix();
-  }
-}
 
-async function prepareEnvironmentPosix(): Promise<boolean> {
-
-  assert((process.platform !== 'win32'));
-
-  // check for which R override
-  let rWhichRPath = new FilePath();
-  const whichROverride = getenv('RSTUDIO_WHICH_R');
-  if (whichROverride) {
-    rWhichRPath = new FilePath(whichROverride);
+  try {
```

Comment:
I'd avoid using `Promise<boolean>`, and instead make this `Promise<void>`. See my message in Slack channel about this. There are other instances of this pattern, I'm going to go back and clean them up.

That eliminates the need for a try/catch here. An `async` function takes care of catching thrown exceptions and performing equivalent of `Promise.reject(e)` for you.

This does require callers to use promise-semantics. If using `await` then the error is made into an exception which will need to be caught, or if using promises directly, should be handled in the `.catch()` callback.

This does bring up a need for us to chat about what patterns we want to use here (exceptions, etc.) so we can agree on conventions to follow.

Reply:
Also, we'll want to look more closely at whether we get any value from the async-ness in these routines (yes, it was my fault they were to begin with, mainly me playing around and learning, honestly).

For a desktop app like this, especially startup code, might not be worth the complexity, but haven't gone back to reason that through.